src/components/CanvasSaveButton/CanvasSaveButton.js

```diff
@@ -343,18 +410,19 @@ class CanvasSaveButton extends PureComponent {
           clip.h,
         );
 
-        // Copyright
-        if (extraData && extraData.copyright && extraData.copyright.text) {
-          this.drawCopyright(destContext, clip);
-        }
-
         // North arrow
         let p = Promise.resolve();
         if (extraData && extraData.northArrow) {
           p = this.drawNorthArrow(destContext, clip);
         }
 
-        p.then(() => {
+        p.then(res => {
+          const arrowWidth = res || 0;
+          // Copyright
+          if (extraData && extraData.copyright && extraData.copyright.text) {
+            this.drawCopyright(destContext, clip, arrowWidth);
```

Comment:
I think you want to send the maxWidth here. It avoid calculating it in subsequent function.